tests/lib/rules/newline_before_scenario.test.js

```diff
@@ -81,6 +124,66 @@ describe("newline_before_scenario", () => {
                     ),
                 ],
             ],
+            [
+                "more newlines with a tag",
+                "Feature: a feature\n\n\n\n  @tag\n  Scenario: a scenario",
+                [
+                    generateProblem(
+                        { line: 6, column: 3 },
+                        3,
+                        customConfig.option[0]
+                    ),
+                ],
+            ],
+        ])("%s", (_, text, expectedProblems) => {
+            const ast = parser.parse(text);
+            const problems = NewlineBeforeScenario.run(ast, customConfig);
+            expect(problems.length).toEqual(expectedProblems.length);
+            problems.forEach((problem, index) => {
+                expect(problem.location).toEqual(
+                    expectedProblems[index].location
+                );
+                expect(problem.message).toEqual(
+                    expectedProblems[index].message
+                );
+            });
+        });
+    });
+
+    describe("custom option: count tag and comment as newline", () => {
+        const customConfig = {
+            type: "error",
+            option: [2, true],
+        };
+        it.each([
+            [
+                "valid number of new lines",
+                "Feature: a feature\n\n  @tag\n  Scenario: a scenario",
+                [],
+            ],
+            [
+                "no newline",
+                "Feature: a feature\n  @tag\n  Scenario: a scenario",
+                [
+                    generateProblem(
+                        { line: 3, column: 3 },
+                        0,
+                        customConfig.option[0]
+                    ),
+                ],
+            ],
+            [
+                "more newlines",
+                "Feature: a feature\n\n\n\n  @tag\n  Scenario: a scenario",
+                [
+                    generateProblem(
+                        { line: 6, column: 3 },
+                        3,
+                        customConfig.option[0],
+                        "Expected 1 newline(s) before Scenario but found 3 (including tags and comments)"
```

Comment:
There are 3 empty lines, so I suppose the message is "correct".
But, including tags and comments, there are 4 lines between Feature and Scenario.

And it says "Expected 1 newline(s)" - the rule is to expect 2 "lines" (including tags and comments). In this case there is a tag line, so the net ef - users fect is that only 1 newline is expected.

Actually I am not sure what is the best way to report this. So it is OK like this - users will be able to know that they have 2 newlines excess.

Reply:
yeah, the implemented in way that the number always represents newlines